src/ui/liferea_shell.c

```diff
@@ -1355,23 +1356,15 @@ void
 liferea_shell_toggle_visibility (void)
 {
 	GtkWidget *mainwindow = GTK_WIDGET (shell->window);
+	GdkWindow *gdkwindow = gtk_widget_get_window (mainwindow);
 
-	if (gdk_window_get_state (gtk_widget_get_window (mainwindow)) & GDK_WINDOW_STATE_ICONIFIED) {
-		/* The window is either iconified, or on another workspace */
-		/* Raise it in one click */
-		if (gtk_widget_get_visible (mainwindow)) {
-			liferea_shell_save_position ();
-			gtk_widget_hide (mainwindow);
-		}
+	if (gdk_x11_window_get_desktop (gdkwindow) !=
+	    gdk_x11_screen_get_current_desktop (gdk_window_get_screen (gdkwindow))) {
+		gdk_x11_window_move_to_current_desktop (gdkwindow);
```

Comment:
This is X11 specific.
The [doc suggests](https://developer.gnome.org/gdk3/stable/gdk3-X-Window-System-Interaction.html) to check the backend in use.
I don't have Wayland installed, so I didn't check what happens.
This function is only used in the trayicon plugin for now and the plugin is useless on Wayland, but it shouldn't be specific to this plugin. Or then, we should just move it there, but I think hiding the window might be desirable on Wayland too ?

Reply:
Good catch. I think it'd be easiest to keep this in here vs. the plugin, but I agree it should be wrapped in a check. I also don't have Wayland, so I'm not sure what happens there.

If someone feels strongly that this should be moved into the plugin, I can look into that as well. I agree that hiding the window might be desirable on Wayland, in which case keeping protocol specific bits in here would seem to make the most sense.

Reply:
I went ahead and wrapped the X11 specific APIs as described in the docs.